[Bug]: test_install.py::test_install_qt6_wasm_autodesktop failures

### Bug description

FAILED tests/test_install.py::test_install_qt6_wasm_autodesktop[6.8.0-680-wasm_singlethread-False] - assert 1 == 0
FAILED tests/test_install.py::test_install_qt6_wasm_autodesktop[6.10.1-6101-wasm_singlethread-True] - assert 1 == 0

These appeared with https://github.com/miurahr/py7zr/commit/9ce01f956b6ef1fd3e7d41a3bd28f45844fcd860
With the previous commit, https://github.com/miurahr/py7zr/commit/8f8a6160fdd3fdc2819bd25b5b76c13889dc5b2a, the tests passed.

With the later commit the files in our archive don't have write permission:
```
(aqttest-venv) tsteven4@PEDALDAMNIT:/tmp/tmpp5n9w4xh$ find . -ls
   468147      4 drwx------   3 tsteven4 tsteven4     4096 Apr 21 19:40 .
   596721      4 drwxr-xr-x   3 tsteven4 tsteven4     4096 Apr 21 19:40 ./6.10.1
   596723      4 drwxr-xr-x   3 tsteven4 tsteven4     4096 Apr 21 19:40 ./6.10.1/wasm_singlethread
   596727      4 drwxr-xr-x   2 tsteven4 tsteven4     4096 Apr 21 19:40 ./6.10.1/wasm_singlethread/bin
   612228      4 ----------   1 tsteven4 tsteven4       28 Apr 21 19:40 ./6.10.1/wasm_singlethread/bin/qmake6
   757431      4 ----------   1 tsteven4 tsteven4       29 Apr 21 19:40 ./6.10.1/wasm_singlethread/bin/qtpaths
   597880      4 ----------   1 tsteven4 tsteven4       11 Apr 21 19:40 ./6.10.1/wasm_singlethread/bin/target_qt.conf
  1100165      4 ----------   1 tsteven4 tsteven4       30 Apr 21 19:40 ./6.10.1/wasm_singlethread/bin/qtpaths6
   612224      4 ----------   1 tsteven4 tsteven4       27 Apr 21 19:40 ./6.10.1/wasm_singlethread/bin/qmake
```

With the earlier commit:
```
(aqttest-venv) tsteven4@PEDALDAMNIT:~/work/aqtinstall$ find /tmp/tmphbefr4n6/ -ls
   392245      4 drwx------   3 tsteven4 tsteven4     4096 Apr 21 19:48 /tmp/tmphbefr4n6/
   596727      4 drwxr-xr-x   4 tsteven4 tsteven4     4096 Apr 21 19:48 /tmp/tmphbefr4n6/6.10.1
  1100203      4 drwxr-xr-x   4 tsteven4 tsteven4     4096 Apr 21 19:48 /tmp/tmphbefr4n6/6.10.1/gcc_64
  1100210      4 drwxr-xr-x   2 tsteven4 tsteven4     4096 Apr 21 19:48 /tmp/tmphbefr4n6/6.10.1/gcc_64/lib
  1100204      4 drwxr-xr-x   2 tsteven4 tsteven4     4096 Apr 21 19:48 /tmp/tmphbefr4n6/6.10.1/gcc_64/bin
  1100205      0 -rw-r--r--   1 tsteven4 tsteven4        0 Apr 21 19:48 /tmp/tmphbefr4n6/6.10.1/gcc_64/bin/dummy
  1100195      4 -rw-r--r--   1 tsteven4 tsteven4       18 Apr 21 19:48 /tmp/tmphbefr4n6/6.10.1/gcc_64/bin/qt.conf
   597880      4 drwxr-xr-x   3 tsteven4 tsteven4     4096 Apr 21 19:48 /tmp/tmphbefr4n6/6.10.1/wasm_singlethread
   612224      4 drwxr-xr-x   2 tsteven4 tsteven4     4096 Apr 21 19:48 /tmp/tmphbefr4n6/6.10.1/wasm_singlethread/bin
  1100165      4 -rwxr-xr-x   1 tsteven4 tsteven4       28 Apr 21 19:48 /tmp/tmphbefr4n6/6.10.1/wasm_singlethread/bin/qmake6
  1100191      4 -rwxr-xr-x   1 tsteven4 tsteven4       29 Apr 21 19:48 /tmp/tmphbefr4n6/6.10.1/wasm_singlethread/bin/qtpaths
   612228      4 -rw-r--r--   1 tsteven4 tsteven4       11 Apr 21 19:48 /tmp/tmphbefr4n6/6.10.1/wasm_singlethread/bin/target_qt.conf
  1100192      4 -rwxr-xr-x   1 tsteven4 tsteven4       30 Apr 21 19:48 /tmp/tmphbefr4n6/6.10.1/wasm_singlethread/bin/qtpaths6
   757431      4 -rwxr-xr-x   1 tsteven4 tsteven4       27 Apr 21 19:48 /tmp/tmphbefr4n6/6.10.1/wasm_singlethread/bin/qmake
```

It seems like that py7zr commit broke writestr.

### Expected behavior

all tests to pass

### aqt and python version

aqtinstall(aqt) v3.3.1.dev106 on Python 3.12.3 [CPython GCC 13.3.0]

### Operating System

Linux/Unix

### Relevant log output

```shell
INFO    : Patching /tmp/tmp3s6hi5wh/6.10.1/wasm_singlethread/bin/qmake
ERROR   : Updater caused an IO error: [Errno 13] Permission denied: '/tmp/tmp3s6hi5wh/6.10.1/wasm_singlethread/bin/qmake'
```

--- a/tests/test_install.py
+++ b/tests/test_install.py
@@ -11,7 +11,7 @@ import textwrap
 from dataclasses import dataclass
 from datetime import datetime
 from pathlib import Path
-from tempfile import NamedTemporaryFile, TemporaryDirectory
+from tempfile import TemporaryDirectory
 from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple
 
 import py7zr
@@ -1512,7 +1512,7 @@ def test_install(
         ("6.10.1", "6101", "wasm_singlethread", True),
     ],
 )
-def test_install_qt6_wasm_autodesktop(monkeypatch, capsys, version, str_version, wasm_arch, extract_target):
+def test_install_qt6_wasm_autodesktop(monkeypatch, capsys, tmp_path, version, str_version, wasm_arch, extract_target):
     """Test installing Qt 6.8 WASM with autodesktop, which requires special handling for addons"""
 
     if sys.platform.startswith("linux"):
@@ -1596,14 +1596,9 @@ def test_install_qt6_wasm_autodesktop(monkeypatch, capsys, version, str_version,
                     )
 
                 for filepath, content in basic_files.items():
-                    # use delete_on_close=False instead of delete=False, and context manager when python >= 3.12.
-                    mockfile = NamedTemporaryFile(delete=False)
-                    try:
-                        mockfile.write(content.encode("utf-8"))
-                        mockfile.close()
-                        archive.write(mockfile.name, filepath)
-                    finally:
-                        os.unlink(mockfile.name)
+                    mockfile = tmp_path / Path(filepath).name
+                    mockfile.write_text(content, encoding="utf-8")
+                    archive.write(mockfile, filepath)
 
     # Setup mocks
     monkeypatch.setattr("aqt.archives.getUrl", mock_get_url)

--- a/tests/test_install.py
+++ b/tests/test_install.py
@@ -11,7 +11,7 @@ import textwrap
 from dataclasses import dataclass
 from datetime import datetime
 from pathlib import Path
-from tempfile import TemporaryDirectory, NamedTemporaryFile
+from tempfile import NamedTemporaryFile, TemporaryDirectory
 from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple
 
 import py7zr
@@ -1596,10 +1596,14 @@ def test_install_qt6_wasm_autodesktop(monkeypatch, capsys, version, str_version,
                     )
 
                 for filepath, content in basic_files.items():
-                    with NamedTemporaryFile(delete_on_close=False) as mockfile:
+                    # use delete_on_close=False instead of delete=False, and context manager when python >= 3.12.
+                    mockfile = NamedTemporaryFile(delete=False)
+                    try:
                         mockfile.write(content.encode("utf-8"))
                         mockfile.close()
                         archive.write(mockfile.name, filepath)
+                    finally:
+                        os.unlink(mockfile.name)
 
     # Setup mocks
     monkeypatch.setattr("aqt.archives.getUrl", mock_get_url)

--- a/tests/test_install.py
+++ b/tests/test_install.py
@@ -11,7 +11,7 @@ import textwrap
 from dataclasses import dataclass
 from datetime import datetime
 from pathlib import Path
-from tempfile import TemporaryDirectory
+from tempfile import TemporaryDirectory, NamedTemporaryFile
 from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple
 
 import py7zr
@@ -1582,10 +1582,10 @@ def test_install_qt6_wasm_autodesktop(monkeypatch, capsys, version, str_version,
                     basic_files.update(
                         {
                             f"{prefix}bin/target_qt.conf": "Prefix=...\n",  # Basic config
-                            f"{prefix}bin/qmake{extension}": "echo Mock qmake{extension}\n",
-                            f"{prefix}bin/qmake6{extension}": "echo Mock qmake6{extension}\n",
-                            f"{prefix}bin/qtpaths{extension}": "echo Mock qtpaths{extension}\n",
-                            f"{prefix}bin/qtpaths6{extension}": "echo Mock qtpaths6{extension}\n",
+                            f"{prefix}bin/qmake{extension}": f"echo Mock qmake{extension}\n",
+                            f"{prefix}bin/qmake6{extension}": f"echo Mock qmake6{extension}\n",
+                            f"{prefix}bin/qtpaths{extension}": f"echo Mock qtpaths{extension}\n",
+                            f"{prefix}bin/qtpaths6{extension}": f"echo Mock qtpaths6{extension}\n",
                         }
                     )
                 else:
@@ -1596,7 +1596,10 @@ def test_install_qt6_wasm_autodesktop(monkeypatch, capsys, version, str_version,
                     )
 
                 for filepath, content in basic_files.items():
-                    archive.writestr(content.encode("utf-8"), filepath)
+                    with NamedTemporaryFile(delete_on_close=False) as mockfile:
+                        mockfile.write(content.encode("utf-8"))
+                        mockfile.close()
+                        archive.write(mockfile.name, filepath)
 
     # Setup mocks
     monkeypatch.setattr("aqt.archives.getUrl", mock_get_url)

